src/resources/js/dropdowns.js

```diff
@@ -83,38 +83,76 @@ var tribe_dropdowns = tribe_dropdowns || {};
 		return result;
 	};
 
-	obj.init_selection = function ( element, callback ) {
-		var $select = element,
-			args = $select.data( 'dropdown' ),
-			data = [],
-			is_multiple = $select.is( '[data-multiple]' );
-
-		$( element.val().split( args.regexSplit ) ).each( function () {
-			var item = { id: this, text: this };
-			if ( args.data.length > 0  && _.isObject( args.data[0] ) ) {
-				var _item = _.where( args.data, { text: this.valueOf() } );
-				if ( _item.length > 0 ){
-					item = _item[0];
-				} else {
-					_item = _.where( args.data, { id: parseInt( this.valueOf(), 10 ) } );
-					if ( _item.length > 0 ){
-						item = _item[0];
-					}
-				}
-			}
-
-			if ( is_multiple ) {
-				data.push( item );
-			} else {
-				callback( item );
+	/**
+	 * If the element used as the basis of a dropdown specifies one or more numeric/text
+	 * identifiers in its val attribute, then use those to preselect the appropriate options.
+	 *
+	 * @param {object}   $select
+	 * @param {function} make_selection
+	 */
+	obj.init_selection = function( $select, make_selection ) {
+		var is_multiple    = $select.is( '[data-multiple]' ),
+		    options        = $select.data( 'dropdown' ),
+		    current_values = $select.val().split( options.regexSplit ),
+		    selected_items = [];
+
+		$( current_values ).each( function() {
```

Comment:
So I was thinking we could use `_.reduce(list, iteratee, [memo], [context])` to apply a method somewhat like `find_item`.